loaders/authentication.js

```diff
@@ -0,0 +1,47 @@
+const fetch = require('node-fetch');
+const { get } = require('../routes');
+const dotenv = require('dotenv').config();
+
+const URL = 'https://sigarra.up.pt/feup/pt/mob_val_geral.autentica?'
+
+let urlRequest = URL + 'pv_login=' + process.env.TTS_USER + '&pv_password=' + process.env.TTS_PWD;
+
+//let cookie = undefined;
+
+async function getCookie(){
+   try {
+        const response = await fetch(urlRequest, {
+            method: 'POST'
+        });
+        if(response.ok){
+            const res = await response.json();
+            return res;
+        }
+        throw new Error('Request Failed!');
+
+   }catch (error){
+       console.log(error);
+   }
+}
+
+
+async function returnCookie(){
```

Comment:
This can be rewritten more simply as:
```
if (cookie === null)
    cookie = getCookie();
return cookie;
```

Reply:
I would suggest setting cookie to null in the beggining, instead of leaving it undefined as is now.
Basically uncomment line 9 and make cookie = null.

Reply:
But should we keep the returnCookie function?

Reply:
I think it makes sense to keep it

Reply:
ok